src/cloud_provider/scaleway/router.rs

```diff
@@ -145,6 +145,51 @@ impl Service for Router {
             .map(|x| x.unwrap())
             .collect::<Vec<_>>();
 
+        let kubernetes_config_file_path = kubernetes.config_file_path();
+
+        match kubernetes_config_file_path {
+            Ok(kubernetes_config_file_path_string) => {
+                // Default domain
+                let external_ingress_hostname_default = crate::cmd::kubectl::kubectl_exec_get_external_ingress_hostname(
+                    kubernetes_config_file_path_string.as_str(),
+                    "nginx-ingress",
+                    "app=nginx-ingress,component=controller",
+                    kubernetes.cloud_provider().credentials_environment_variables(),
+                );
+
+                match external_ingress_hostname_default {
+                    Ok(external_ingress_hostname_default_result) => match external_ingress_hostname_default_result {
+                        Some(hostname) => context.insert("external_ingress_hostname_default", hostname.as_str()),
```

Comment:
OPT: If you remove the external target annotation in the ingress, you don't need all this code.
It will be done automatically for you by the ingress controller

Reply:
Done